packages/headless/src/features/triggers/triggers-state.ts

```diff
@@ -48,6 +48,15 @@ export interface TriggerState {
    * The notifications to display to the user after receiving notification triggers.
    */
   notifications: string[];
+
+  /**
+   * The modification that should be applied as a result of a query trigger.
+   */
+  queryModification: {
```

Comment:
I think it would make more sense for `queryModification` to be `undefined` when no correction occured. Perhaps `ignore` could be extracted out of this object if that's the case.

Reply:
Whenever possible I think it's better to have empty object instead of undefined object to represent an empty state.

Having `undefined` in some places in the state has a much bigger impact on downstream code that needs to check for null ref everywhere, vs having an empty object keeps the complexity local to the state + slice code.